### PR TITLE
Task Feat: View All link & Icon in header of widgets

### PR DESCRIPTION
**Pull request by shitalkakade24**

### CODE REVIEW

**Comment by thite-amol** [comment]
@coolbung Can you please check this

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js [hunk 1]
```diff
@@ -37,11 +38,15 @@ var TJDashboardUI = {
 					value.params = JSON.parse(value.params);
 				}
 
-				if(value.params.color!=undefined && value.params.color.length!=0){
+				if(value.params.color){
```
Comment (coolbung) [inline src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: Are you sure the params will always contain color ? You should change this to

`if (typeof(value.params.color) != 'undefined' && value.params.color)`

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js [hunk 2]
```diff
@@ -37,11 +38,15 @@ var TJDashboardUI = {
...
 					colorClass=value.params.color;
 				}
 
-				jQuery('<div class="col-xs-' +value.size+'"><div class="widget-data panel '+colorClass+'"><div class="widget-title panel-heading"><b>'+value.title+'</b></div><div data-dashboard-widget-id="'+value.dashboard_widget_id+'" id="dashboard-widget-'+value.dashboard_widget_id+'" class=""></div></div></div>').appendTo('.dashboard-widget-row-'+j);
+				if(value.params.icon){
```
Comment (coolbung) [inline src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: Same as above comment

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js [hunk 3]
```diff
@@ -37,11 +38,15 @@ var TJDashboardUI = {
 					value.params = JSON.parse(value.params);
```
Comment (coolbung) [inline src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: JSON.parse throws an exception if the value is not valid JSON. Are you sure there will always be valid JSON inside ?

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js (hunk 1 above)
Comment (shitalkakade24) [reply src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: Are you sure the params will always contain color ? You should change this to: 
Default color value is set before if

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js (hunk 2 above)
Comment (shitalkakade24) [reply src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: it is also defined. This changes told by Amol Sir.

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js (hunk 3 above)
Comment (shitalkakade24) [reply src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: changes done

File: src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js (hunk 2 above)
Comment (coolbung) [reply src/components/com_tjdashboard/site/assets/js/tjDashboardUI.js]: You have set the default value of `colorClass` and `icon` variables. I'm talking about the icon and color attributes in the config JSON.

**Pull request merged**